File: String compression/ans.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int vcount = 0, ecount = 1;
        char element = chars[0];
        for (int i = 1 ; i < chars.size(); i++) {
            if (chars[i] == element) {
                ecount++;
            } else {
                chars[vcount++] = element;
                if (ecount > 1) {
                    string countStr = to_string(ecount);
                    for (char ch : countStr) {
                        chars[vcount++] = ch;
                    }
                }
                element = chars[i];
                ecount = 1;
            }
        }


        chars[vcount++] = element;
        if (ecount > 1) {
            string countStr = to_string(ecount);
            for (char ch : countStr) {
                chars[vcount++] = ch;
            }
        }
        return vcount;

        
    }
};

File: String compression/ans.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        acount = 0
        element = chars[0]
        count = 1

        for i in range(1, len(chars)):
            if chars[i] == element:
                count += 1
            else:
                chars[acount] = element
                acount += 1
                if count > 1:
                    for c in str(count):
                        chars[acount] = c
                        acount += 1
                element = chars[i]
                count = 1

        chars[acount] = element
        acount += 1
        if count > 1:
            for c in str(count):
                chars[acount] = c
                acount += 1

        return acount


        
